### PR TITLE
refactor: switching more resources over to using ID Parsers

### PR DESCRIPTION
**Pull request by tombuildsstuff**

### CODE REVIEW

File: internal/services/loganalytics/migration/saved_search_v0_to_v1.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+type SavedSearchV0ToV1 struct{}
+
+func (SavedSearchV0ToV1) Schema() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"log_analytics_workspace_id": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			ValidateFunc:     validate.LogAnalyticsWorkspaceID,
+			DiffSuppressFunc: suppress.CaseDifference,
```
Comment (stephybun) [inline internal/services/loganalytics/migration/saved_search_v0_to_v1.go:27]: Can't we remove these?

File: internal/services/loganalytics/migration/saved_search_v0_to_v1.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		},
+
+		"name": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			DiffSuppressFunc: suppress.CaseDifference,
```
Comment (stephybun) [inline internal/services/loganalytics/migration/saved_search_v0_to_v1.go:34]: Same for these as well as for the properties below.

File: internal/services/loganalytics/migration/saved_search_v0_to_v1.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		oldId := rawState["id"].(string)
+
+		id, err := parse.LogAnalyticsSavedSearchID(fmt.Sprintf("/%s", strings.TrimPrefix(oldId, "/")))
+		if err != nil {
+			return rawState, err
+		}
+
+		newId := id.ID()
+		log.Printf("[DEBUG] Updating ID from %q to %q..", oldId, newId)
```
Comment (stephybun) [inline internal/services/loganalytics/migration/saved_search_v0_to_v1.go:91]:
```suggestion
		log.Printf("[DEBUG] Updating ID from %q to %q", oldId, newId)
```

**Review by stephybun — APPROVED** [review]
Left some minor comments, otherwise LGTM 👍🏼

File: internal/services/loganalytics/migration/saved_search_v0_to_v1.go (hunk 2 above)
Comment (tombuildsstuff) [reply internal/services/loganalytics/migration/saved_search_v0_to_v1.go:34]: it's a state migration, so it's worth leaving this in for now

File: internal/services/loganalytics/migration/saved_search_v0_to_v1.go (hunk 3 above)
Comment (tombuildsstuff) [reply internal/services/loganalytics/migration/saved_search_v0_to_v1.go:91]: this matches the other state migrations fwiw, leaving this as-is for now

**Pull request merged**